src/energyplus/ForwardTranslator/ForwardTranslateElectricEquipment.cpp

```diff
@@ -57,23 +57,25 @@ namespace openstudio {
 namespace energyplus {
 
   boost::optional<IdfObject> ForwardTranslator::translateElectricEquipment(ElectricEquipment& modelObject) {
-    IdfObject idfObject(openstudio::IddObjectType::ElectricEquipment);
-    m_idfObjects.push_back(idfObject);
+
+    IdfObject idfObject = createRegisterAndNameIdfObject(openstudio::IddObjectType::ElectricEquipment, modelObject);
 
     for (LifeCycleCost lifeCycleCost : modelObject.lifeCycleCosts()) {
       translateAndMapModelObject(lifeCycleCost);
     }
 
     ElectricEquipmentDefinition definition = modelObject.electricEquipmentDefinition();
 
-    idfObject.setString(ElectricEquipmentFields::Name, modelObject.name().get());
-
     boost::optional<Space> space = modelObject.space();
     boost::optional<SpaceType> spaceType = modelObject.spaceType();
     if (space) {
-      boost::optional<ThermalZone> thermalZone = space->thermalZone();
-      if (thermalZone) {
-        idfObject.setString(ElectricEquipmentFields::ZoneorZoneListorSpaceorSpaceListName, thermalZone->name().get());
+      if (m_excludeSpaceTranslation) {
```

Comment:
I expect that in going through this, we'll see this block of code often. I wonder if it can / should be factored out?

Reply:
Yeah I think something should be done to factor it out (used by 10+ functions) but I didn't feel I had the right construct to do it yet